Campus-iOS/GradesComponent/Screen/GradesScreen.swift

```diff
@@ -26,63 +26,73 @@ struct GradesScreen: View {
     }
     
     var body: some View {
-        Group {
-            switch vm.state {
-            case .success(_):
-                VStack {
-                    GradesView(
-                        vm: self.vm
-                    )
-                    .padding(.top, 30)
-                    .refreshable {
-                        await vm.reloadGradesAndAverageGrades(forcedRefresh: true)
+        if (vm.grades.count > 0) {
```

Comment:
please remove that if else clause - applying this before data is fetched, makes no sense

Reply:
it's already in the `success` case, cause the incoming data in the empty case is `[]`, which is considered ti the success case. That's whyI handled the main logic in success

Reply:
please remove it at line 29 and 92